spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/XsuaaToken.java

```diff
@@ -141,7 +143,19 @@ public String getLogonName() {
 	@Override
 	@Nullable
 	public String getClientId() {
-		return getClaimAsString(CLAIM_CLIENT_ID);
+		String clientId = getClaimAsString(AUTHORIZATION_PARTY);
+		if (clientId == null || clientId.trim().isEmpty()) {
+			List<String> audiences = getAudience();
+
+			if (audiences != null && audiences.size() == 1) {
+				return audiences.get(0);
+			} else {
+				logger.error("Couldn't get client id. Invalid authorized party or audience claims.");
```

Comment:
check for cid (and add a comment that this is only required for support generated tokens in JUnit tests. Finally they should provide "azp" claim instead of "cid" claim) and then als log a WARNING